src/components/cross-section-component.tsx

```diff
@@ -65,6 +67,14 @@ export class CrossSectionComponent extends BaseComponent<IProps, IState>{
 
   public componentDidUpdate(prevProps: IProps) {
     this.recomputeMetrics();
+
+    if (this.props.isSelectingCrossSection !== prevProps.isSelectingCrossSection) {
+      this.stores.setCrossSectionSelectorVisibility(this.props.isSelectingCrossSection);
+    }
+
+    if (!this.stores.showCrossSectionSelector && this.state.isSelecting) {
+      this.setState({isSelecting: false});
```

Comment:
We are managing `isSelecting` in local state, how does this overlap with with `props.isSelectingCrossSection`.   

I feel like we can simplify things. I would try to manage these items in one place. Probably the application state. (Stores)